{{ cookiecutter.package_name }}/.github/workflows/tox-tests.yml

```diff
@@ -0,0 +1,104 @@
+
+name: Python Tests
+
+on: [push, pull_request]
+
+jobs:
+
+  # Set up matrix to run tox tests across lists of os, python version, and tox environment
+  matrix_tests:
+    runs-on: ${{ matrix.os }}
+    if: "!contains(github.event.head_commit.message, '[ci skip]')"  # github actions doesn't yet support this natively so add here
+    strategy:
+      matrix:
+        # Github actions supports ubuntu, windows, and macos virtual environments:
+        # https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
+        #
+        # Only run on ubuntu by default, but can add other os's to the test matrix here.
+        # For example -- os: [ubuntu-latest, macos-latest, windows-latest]
+        os: [ubuntu-latest]
+
+        # Test python 3.7 and 3.8 by default.
+        python-ver: [7, 8]
+
+        # Specify which tox environments to test in this list.
+        tox-env: [cov, alldeps, devdeps, astropylts]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up python 3.${{ matrix.python-ver }} with tox environment py3${{ matrix.python-ver }}-${{ matrix.tox-env }} on ${{ matrix.os }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.${{ matrix.python-ver }}
+    - name: Install base dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install tox
+    - name: Test with tox
+      run: |
+        tox -e py3${{ matrix.python-ver }}-${{ matrix.tox-env }}
+    #
+    # # This is an example of how to upload coverage to codecov
+    # - name: Upload coverage to codecov
+    #   if: matrix.tox-env == 'cov' && matrix.python-ver == '8'
+    #   uses: codecov/codecov-action@v1
+    #   with:
+    #     token: ${{ secrets.CODECOV }}
+    #     file: ./coverage.xml
+    #     fail_ci_if_error: true
+
+  # Test building of documents
+  doc_build:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[ci skip]')"
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python to build docs with sphinx
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: Install base dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install tox
+    - name: Testing building of docs using tox
+      run: |
+        tox -e build_docs
+
+  # Test links in documents
+  doc_test:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[ci skip]')"
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python to build docs with sphinx
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: Install base dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install tox
+    - name: Check links in docs using tox
+      run: |
+        tox -e linkcheck
```

Comment:
From my experience, checking links every push even isn't that useful. I like to put this in cron job (daily or weekly is up to the maintainer).

Reply:
good point. when editing docs, i've found the link check test useful for catching typos or bad redirects. however, that can be done locally easy enough so a scheduled test for CI is probably best for most usage cases.

Reply:
i did move the link check to a cron scheduled workflow as an example of how to do so.